tests/conftest.py

```diff
@@ -18,6 +25,40 @@ def mock_tokenizer() -> Tokenizer:
     return tokenizer
 
 
+@pytest.fixture
+def mock_berttokenizer() -> BertTokenizerFast:
+    """Load the real BertTokenizerFast from the provided tokenizer.json file."""
+    tokenizer_path = Path("tests/data/test_tokenizer/tokenizer.json")
+    return BertTokenizerFast(tokenizer_file=str(tokenizer_path))
+
+
+@pytest.fixture
+def mock_transformer() -> AutoModel:
+    """Create a mock transformer model."""
+
+    class MockPreTrainedModel:
+        def __init__(self) -> None:
+            self.device = "cpu"
+
+        def to(self, device: str) -> MockPreTrainedModel:
+            self.device = device
+            return self
+
+        def __call__(self, *args: Any, **kwargs: Any) -> Any:
```

Comment:
One cool addition could be to either make the configurable, or make the seed of the rand depend on the input data. This makes tests a lot more reproducible and predictable. Anyway, just a small thing.

Something like an MD5 hash of the input_ids as a tuple, or maybe just the sum? Idk